lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +133,76 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+  // TODO: how would we represent multiplicity of roots
+  // should we return the root x times or just return it once
+*/
+
+function getSolutionsAndSymbol (equation) {
+  const left = equation.leftNode;
+  const rightNode = Node.Creator.constant(0);
+
+  const solutions = [];
+
+  let symbol, steps, factorsWithSymbols;
+
+  // If left hand side is a power node and it does not resolve to a constant (e.g. 2^7)
+  // then it is a factor. If left hand side is a multiplication node, return a list of all the valid factors.
+  if (Node.Type.isOperator(left, '^') && !checks.resolvesToConstant(left)){
+    factorsWithSymbols = [left];
+  }
+  else {
+    factorsWithSymbols = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
+  }
+
+  // For each factor, solve the equation factor = 0 for the symbol
+  // If the factor is a power node, solve the equation base = 0 for the symbol
+  for (var f in factorsWithSymbols) {
+    const factor = factorsWithSymbols[f];
+
+    let leftNode;
+
+    if (Node.Type.isOperator(factor, '^')) {
```

Comment:
You could clean up this section by changing the logic as follows:
1. if exponent, then factor = factor.args[0]
2. now check for parenthesis on factor
That way the parenthesis check only happens once